flake.nix

```diff
@@ -16,7 +16,133 @@
+            };
+          };
+          programs.polykey = {
+            enable = mkEnableOption "Enable the per-user Polykey agent.";
+
+            recovery = mkEnableOption "Enable recovery mode for the node. Will try to source from a file under the configuration directory called `recovery-code`.";
+
+            configPath = mkOption {
+              type = with types; uniq str;
+              default = "%h/.config/polykey";
+              description = ''
+              The path to each user's Polykey configuration directory.
+              A file called `password` with your desired password must be placed inside of this directory to set the node password.
+              '';
+            };
+
+            statePath = mkOption {
+              type = with types; uniq str;
+              default = "%h/.local/share/polykey";
+              description = "The path to each user's Polykey node state directory.";
+            };
+          };
+        };
+        config = mkMerge [
+          (mkIf config.services.polykey.enable {
+            users.groups.polykey = {};
+
+            environment.systemPackages = [
+              self.outputs.packages.${buildSystem}.default
+            ];
+
+            system.activationScripts.makeAgentPaths = ''
+              mkdir -p ${config.services.polykey.configPath}
+              mkdir -p ${config.services.polykey.statePath}
+              chgrp -R polykey ${config.services.polykey.configPath}
+              chgrp -R polykey ${config.services.polykey.statePath}
+              chmod 770 ${config.services.polykey.configPath}
+              chmod 770 ${config.services.polykey.statePath}
+            '';
+
+            systemd.services.polykey = {
+              description = "Polykey Agent";
+              wantedBy = [ "multi-user.target" ];
+              after = [ "network.target" ];
+              serviceConfig = {
+                User = "root";
+                Group = "polykey";
+                PermissionsStartOnly = true;
+                LoadCredential = [
+                  "password:${config.services.polykey.configPath}/password"
+                ];
+                ExecStartPre = ''
+                  -${self.outputs.packages.${buildSystem}.default}/bin/polykey \
+                  -pf ''${CREDENTIALS_DIRECTORY}/password \
+                  -np ${config.services.polykey.statePath} \
+                  bootstrap  ${lib.optionalString config.services.polykey.recovery '' -rcf ${config.services.polykey.configPath}/recovery-code''}\
+                  -rcof ${config.services.polykey.configPath}/recovery-code
+                '';
+                ExecStart = ''
+                  ${self.outputs.packages.${buildSystem}.default}/bin/polykey \
+                  -pf ''${CREDENTIALS_DIRECTORY}/password \
+                  -np ${config.services.polykey.statePath} \
+                  agent start \
+                  -rcof ${config.services.polykey.configPath}/recovery-code
+                '';
```

Comment:
Two things, 

1. Is the `-` in front of `-${self.outputs.packages.${buildSystem}.default}/bin/polykey` meant to be there?
2. Us the full sized option name of `--password-file` and `--node-path` rather than the shortened `-pf` or `-np`. It's better to be explicit here. Make sure all options and flags use the full names.

Reply:
1. The `-` tells Systemd to ignore if the command fails, so if the bootstrap command fails (e.g. the state directory already exists; if the user reboots the service) it won't pull down the entire service with it, and it will continue with executing the `ExecStart` command.
2. That sounds good, I'll implement that now.